.github/workflows/build-git-annex.yaml

```diff
@@ -107,20 +182,44 @@ jobs:
           git annex test
 
   test-annex-more:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     needs: build-package
     strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
       fail-fast: false
 
     steps:
-      - name: Download git-annex package
+      - name: Download git-annex package (Ubuntu)
         uses: actions/download-artifact@v2-preview
         with:
           name: git-annex-debianstandalone-packages
+        if: ${{ runner.os == 'Linux' }}
 
-      - name: Install git-annex package
+      - name: Install git-annex package (Ubuntu)
         run: |
           sudo dpkg -i git-annex*.deb
+        if: ${{ runner.os == 'Linux' }}
+
+      - name: Download git-annex package (macOS)
+        uses: actions/download-artifact@v2-preview
+        with:
+          name: git-annex-macos-dmg
+        if: ${{ runner.os == 'macOS' }}
+
+      - name: Install git-annex package (macOS)
+        run: |
+          hdiutil attach git-annex.dmg
+          rsync -a /Volumes/git-annex/git-annex.app /Applications/
+          hdiutil detach /Volumes/git-annex/
+        if: ${{ runner.os == 'macOS' }}
+
+      - name: Add git-annex to PATH (macOS)
+        run: |
+          echo "::add-path::/Applications/git-annex.app/Contents/MacOS"
+        if: ${{ runner.os == 'macOS' }}
```

Comment:
I think it would become more concise and less likely to become buggy if instead of per OS step, have a step and then `case $${runnes.os} in ...esac`.  That would make it easier to spot in which step for which OS commands differ or what is missing.  And then make all steps into matrix across all supported OSes

Reply:
Done.